File: src/main/java/dev/noyzys/scheduler/api/structure/ScheduledFutureService.java
package dev.noyzys.scheduler.api.structure;

import dev.noyzys.scheduler.api.SchedulerTask;
import org.jetbrains.annotations.NotNull;

import java.util.concurrent.RunnableScheduledFuture;

/*
 * @author: NoyZys on 11:56, 30.08.2020
 */

/**
 * An Interface this represents a task scheduled future.
 * Written api version in: Java
 */
public interface ScheduledFutureService<V> extends RunnableScheduledFuture<V> {

    /**
     * Scheduled task related to this [RunnableScheduledFuture]
     */
    @NotNull
    SchedulerTask task();
}

File: src/main/kotlin/dev/noyzys/sheduler/api/structure/ScheduledFutureService.kt
package dev.noyzys.sheduler.api.structure

import dev.noyzys.sheduler.api.SchedulerTask
import java.util.concurrent.RunnableScheduledFuture

/*
 * @author: NoyZys on 10:57, 30.08.2020
 */

/**
 * An Interface this represents a task scheduled future.
 * Written api version in: Kotlin
 */
interface ScheduledFutureService<V> : RunnableScheduledFuture<V> {

    /**
     * Scheduled task related to this [RunnableScheduledFuture]
     */
    val task: SchedulerTask
}